sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/PartitionKeyBasedBloomFilter.java

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.implementation.guava25.hash.BloomFilter;
+import com.azure.cosmos.implementation.guava25.hash.Funnel;
+import com.azure.cosmos.implementation.guava25.hash.PrimitiveSink;
+import com.azure.cosmos.implementation.routing.PartitionKeyInternal;
+import com.azure.cosmos.implementation.routing.PartitionKeyInternalHelper;
+import com.azure.cosmos.models.PartitionKeyDefinition;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class PartitionKeyBasedBloomFilter {
```

Comment:
Please add some comments about why the bloom filter and what this class holds.

Reply:
Fixed in next iteration.